### PR TITLE
Added CLI recovery information when customers use environment variables & secrets for git based workflows

### PR DESCRIPTION
**Pull request by renebrandel**
_Issue #, if available:_

_Description of changes:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by siegerts — COMMENTED** [review]
Thoughts on highlighting these blocks and/or adding in an example error message for search results? Also, somehow referenced/linked in the Team Environments page as a consideration.

**Review by jhockett — APPROVED** [review]
LGTM

File: docs/cli/function/env-vars.md [hunk 1]
```diff
@@ -50,3 +50,10 @@ When creating a new Amplify environment using `amplify env add --yes`, Amplify C
 
 In multi-environment workflows, you may have added a new environment variable in one Amplify environment and then checked out a different Amplify environment. In this case, on the next `amplify push`, Amplify CLI will detect that there is a new environment variable that does not have a value specified in the current environment and prompt for one.
 Running `amplify push --yes` in this case will fail with a message explaining the missing environment variable values.
+
+In **git-based** multi-environment workflows, you may run into errors during deployment. For example, this happens when you add an environment variable in `envA` (corresponding to a git branch `branchA`), then `amplify checkout envB` and `git checkout branchB` and `git merge` branchA into branchB. Upon pushing `envB`, Amplify CLI detects that a new environment variable has been added but can't infer a value for it. To resolve this issue, run the following commands in the terminal:
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:54]:
```suggestion
In [**git-based** multi-environment workflows](cli/teams/overview.md), you may run into errors during deployment. For example, this happens when you add an environment variable in `envA` (corresponding to a git branch `branchA`), then `amplify checkout envB` and `git checkout branchB` and `git merge` branchA into branchB. Upon pushing `envB`, Amplify CLI detects that a new environment variable has been added but can't infer a value for it. To resolve this issue, run the following commands in the terminal:
```

File: docs/cli/function/secrets.md [hunk 2]
```diff
@@ -69,3 +69,10 @@ When creating a new Amplify environment using `amplify env add`, Amplify CLI ask
 When creating a new Amplify environment using `amplify env add --yes`, Amplify CLI will apply all secret values from the current environment to the new environment.
 
 In multi-environment workflows, you may have added a new secret in one Amplify environment and then checked out a different Amplify environment. In this case, on the next `amplify push`. Amplify CLI will detect that there is a new secret that does not have a value specified in the current environment and prompt for one. Running `amplify push --yes` in this case will fail with a message explaining the missing secret values.
+
+In **git-based** multi-environment workflows, you may run into errors during deployment. For example, this happens when you add a secret in `envA` (corresponding to a git branch `branchA`), then `amplify checkout envB` and `git checkout branchB` and `git merge` branchA into branchB. Upon pushing `envB`, Amplify CLI detects that a new secret has been added but can't infer a value for it. To resolve this issue, run the following commands in the terminal:
```
Comment (renebrandel) [inline docs/cli/function/secrets.md:73]:
```suggestion
In [**git-based** multi-environment workflows](~/cli/teams/overview.md), you may run into errors during deployment. For example, this happens when you add a secret in `envA` (corresponding to a git branch `branchA`), then `amplify checkout envB` and `git checkout branchB` and `git merge` branchA into branchB. Upon pushing `envB`, Amplify CLI detects that a new secret has been added but can't infer a value for it. To resolve this issue, run the following commands in the terminal:
```

**Review by siegerts — APPROVED** [review]
LGTM

**Pull request merged**